pkg/kn/commands/source/ping/create.go

```diff
@@ -76,21 +70,8 @@ func NewPingCreateCommand(p *commands.KnParams) *cobra.Command {
 			}
 			ceOverridesToRemove := util.ParseMinusSuffix(ceOverridesMap)
 
-			var data, dataBase64 string
-
-			switch updateFlags.encoding {
-			case "":
-				if _, err := base64.StdEncoding.DecodeString(updateFlags.data); err == nil {
-					dataBase64 = updateFlags.data
-				} else {
-					data = updateFlags.data
-				}
-			case base64Encoding:
-				dataBase64 = updateFlags.data
-			case textEncoding:
-				data = updateFlags.data
-			default:
-				err := fmt.Errorf("invalid value: %v", updateFlags.encoding)
+			data, dataBase64, err := getDataFields(&updateFlags)
+			if err != nil {
 				fmt.Fprintf(cmd.OutOrStderr(), "Ping source creation failed with error: %v", err)
```

Comment:
The message format should be probably aligned with the rest of sources.
E.g. see: https://github.com/knative/client/blob/main/pkg/kn/commands/source/container/create.go#L64-L68